clients/src/main/java/org/apache/kafka/clients/consumer/internals/OffsetFetcher.java

```diff
@@ -225,14 +231,21 @@ private void resetPositionsAsync(Map<TopicPartition, Long> partitionResetTimesta
 
             RequestFuture<ListOffsetResult> future = sendListOffsetRequest(node, resetTimestamps, false);
             future.addListener(new RequestFutureListener<ListOffsetResult>() {
+                long attempts = 0L;
                 @Override
                 public void onSuccess(ListOffsetResult result) {
-                    offsetFetcherUtils.onSuccessfulRequestForResettingPositions(resetTimestamps, result);
+                    offsetFetcherUtils.onSuccessfulRequestForResettingPositions(
+                            resetTimestamps,
+                            result,
+                            retryBackoff.backoff(attempts++));
```

Comment:
Hmm, does this really do exponential backoff since attempts is 0 for every new request?